cmd/frontend/internal/frontend/metrics.go

```diff
@@ -0,0 +1,260 @@
+	gm := &GatewayMetrics{
+		meter:            meter,
+		metricAttributes: metricAttributes,
+	}
+
+	return gm
+}
+
+func (gm *GatewayMetrics) Set(gateways []*nspAPI.Gateway) {
+	gatewaysMap := map[string]*nspAPI.Gateway{}
+
+	for _, gateway := range gateways {
+		gatewaysMap[fmt.Sprintf("NBR-%s", gateway.Name)] = gateway
+	}
+
+	gm.mu.Lock()
+	defer gm.mu.Unlock()
+	gm.gateways = gatewaysMap
+}
+
+// Collect collects the metrics for the gateways.
+func (gm *GatewayMetrics) Collect() error {
+	if gm.RoutingService == nil {
+		return errors.New("routing service not set for gateway metrics")
+	}
+
+	lp, err := gm.RoutingService.LookupCli()
+	if err != nil {
+		return fmt.Errorf("frontend metrics, failed to RoutingService.LookupCli: %w", err)
+	}
+
+	_, err = gm.meter.Int64ObservableCounter(
+		meridioMetrics.MERIDIO_ATTRACTOR_GATEWAY_ROUTES_IMPORTED,
+		metric.WithUnit("routes"),
+		metric.WithDescription("Counts number of routes imported for a gateway."),
+		metric.WithInt64Callback(func(ctx context.Context, observer metric.Int64Observer) error {
+			return gm.observe(
+				ctx,
+				observer,
+				func(birdStats *BirdStats) int64 {
+					return int64(birdStats.routesImported)
+				},
+				lp,
+			)
+		}),
+	)
+	if err != nil {
+		return fmt.Errorf("frontend metrics, failed to Int64ObservableCounter (%s): %w", meridioMetrics.MERIDIO_ATTRACTOR_GATEWAY_ROUTES_IMPORTED, err)
+	}
+
+	_, err = gm.meter.Int64ObservableCounter(
+		meridioMetrics.MERIDIO_ATTRACTOR_GATEWAY_ROUTES_EXPORTED,
+		metric.WithUnit("routes"),
+		metric.WithDescription("Counts number of routes exported for a gateway."),
+		metric.WithInt64Callback(func(ctx context.Context, observer metric.Int64Observer) error {
+			return gm.observe(
+				ctx,
+				observer,
+				func(birdStats *BirdStats) int64 {
+					return int64(birdStats.routesExported)
+				},
+				lp,
+			)
+		}),
+	)
+	if err != nil {
+		return fmt.Errorf("frontend metrics, failed to Int64ObservableCounter (%s): %w", meridioMetrics.MERIDIO_ATTRACTOR_GATEWAY_ROUTES_EXPORTED, err)
+	}
+
+	_, err = gm.meter.Int64ObservableCounter(
+		meridioMetrics.MERIDIO_ATTRACTOR_GATEWAY_ROUTES_PREFERRED,
+		metric.WithUnit("routes"),
+		metric.WithDescription("Counts number of routes preferred for a gateway."),
+		metric.WithInt64Callback(func(ctx context.Context, observer metric.Int64Observer) error {
+			return gm.observe(
+				ctx,
+				observer,
+				func(birdStats *BirdStats) int64 {
+					return int64(birdStats.routesPreferred)
+				},
+				lp,
+			)
+		}),
+	)
+	if err != nil {
+		return fmt.Errorf("frontend metrics, failed to Int64ObservableCounter (%s): %w", meridioMetrics.MERIDIO_ATTRACTOR_GATEWAY_ROUTES_PREFERRED, err)
+	}
+	return nil
+}
+
+func (gm *GatewayMetrics) observe(ctx context.Context, observer metric.Int64Observer, valueFunc func(*BirdStats) int64, lp string) error {
+	gm.mu.Lock()
+	defer gm.mu.Unlock()
+
+	spaOutput, err := gm.RoutingService.ShowProtocolSessions(ctx, lp, "")
```

Comment:
This captures all the things bird handles as protocol, yet only the ones with the prefix "NBR-" are of interest.
e.g.: ```ShowProtocolSessions(ctx, lp, `NBR-*`)```